torch/ao/quantization/experimental/quantizer.py

```diff
@@ -33,10 +33,13 @@ def quantize(self, tensor2quantize: Tensor):
         result = torch.tensor([])
 
         # map float_to_apot over tensor2quantize elements
-        tensor2quantize = tensor2quantize.apply_(lambda x: float_to_apot(x,
-                                                                         self.quantization_levels,
-                                                                         self.level_indices,
-                                                                         self.alpha))
+        tensor2quantize = tensor2quantize.detach().apply_(lambda x: float_to_apot(x,
+                                                                                  self.quantization_levels,
+                                                                                  self.level_indices,
+                                                                                  self.alpha))
+
+        # convert to APoT int representation for dtype
+        tensor2quantize = tensor2quantize.int()
```

Comment:
can we move this to `float_to_apot`?

Reply:
float_to_apot deals with each element and i need to change the dtype so i think i need it here